### PR TITLE
reduce mem allocations

### PR DESCRIPTION
**Pull request by PSeitz**
remove field mapping allocation per doc

Spotted in heaptrack,  vec of fastfield mappings is created _per_ document
heaptrack

### CODE REVIEW

File: quickwit-doc-mapper/src/default_doc_mapper/field_mapping_entry.rs [hunk 1]
```diff
@@ -45,29 +45,48 @@ pub struct FieldMappingEntry {
...
+            mapping_type,
+            field_entries,
+            fast_field_entries,
+        }
     }
 
     /// Creates a new root [`FieldMappingEntry`].
     pub fn root(mapping_type: FieldMappingType) -> Self {
+        let name = "".to_string();
+        let field_entries = Self::compute_field_entries(name.as_str(), &mapping_type);
```
Comment (fulmicoton) [inline quickwit-doc-mapper/src/default_doc_mapper/field_mapping_entry.rs:71]: wha't's the deal with name?

**Comment by fulmicoton** [comment]
@PSeitz nice catch. Can you double check your code though before sending a CR? I didn't dig much but the  `"".to_string().as_str()` seemed fishy.
@fmassot can you review?

File: quickwit-doc-mapper/src/default_doc_mapper/field_mapping_entry.rs (hunk 1 above)
Comment (PSeitz) [reply quickwit-doc-mapper/src/default_doc_mapper/field_mapping_entry.rs:71]: "" is the name for a root entry

**Comment by PSeitz** [comment]
> @PSeitz nice catch. Can you double check your code though before sending a CR? I didn't dig much but the `"".to_string().as_str()` seemed fishy. @fmassot can you review?

there is no `"".to_string().as_str()`, but something similar where the value is used later

**Comment by fulmicoton** [comment]
@fmassot can you have a look?

**Review by fmassot — APPROVED** [review]
Well spotted!

**Pull request merged**